### PR TITLE
Adding NestedPolynomialFunction as another synthetic function

### PR DESCRIPTION
**Pull request by byte-sculptor**
This change adds another synthetic function to our arsenal of synthetic workloads.

The idea is really simple: create a synthetic function where the first (categorical) parameter -  `polynomial_id` - selects one of the polynomials, and the remaining parameters form the X-vector. The selected polynomial is evaluated for the specified Xs, and that results constitutes the objective function's value. 

Optimizing this function is analogous to optimizing a component with multiple mutually exclusive implementations and no shared parameters. It is a generalization of ThreeLevelQuadratic.

We use the existing PolynomialObjective class to create and evaluate individual polynomials.

### CODE REVIEW

File: source/Mlos.Python/mlos/OptimizerEvaluationTools/SyntheticFunctions/NestedPolynomialObjective.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+import pandas as pd
+
+from mlos.Spaces import CategoricalDimension, ContinuousDimension, Hypergrid, Point, SimpleHypergrid
+from mlos.OptimizerEvaluationTools.ObjectiveFunctionBase import ObjectiveFunctionBase
+from mlos.OptimizerEvaluationTools.SyntheticFunctions.PolynomialObjective import PolynomialObjective
+from mlos.OptimizerEvaluationTools.SyntheticFunctions.PolynomialObjectiveWrapper import PolynomialObjectiveWrapper
+
+class NestedPolynomialObjective(ObjectiveFunctionBase):
+    """ Wraps the PolynomialObjective to provide the interface defined in the ObjectiveFunctionBase.
```
Comment (byte-sculptor) [inline source/Mlos.Python/mlos/OptimizerEvaluationTools/SyntheticFunctions/NestedPolynomialObjective.py:15]: Wrong comment.

File: source/Mlos.Python/mlos/OptimizerEvaluationTools/SyntheticFunctions/NestedPolynomialObjective.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import pandas as pd
+
+from mlos.Spaces import CategoricalDimension, ContinuousDimension, Hypergrid, Point, SimpleHypergrid
+from mlos.OptimizerEvaluationTools.ObjectiveFunctionBase import ObjectiveFunctionBase
+from mlos.OptimizerEvaluationTools.SyntheticFunctions.PolynomialObjective import PolynomialObjective
+from mlos.OptimizerEvaluationTools.SyntheticFunctions.PolynomialObjectiveWrapper import PolynomialObjectiveWrapper
+
+class NestedPolynomialObjective(ObjectiveFunctionBase):
+    """A hierarchical function with multiple nested polynomials.
+
```
Comment (byte-sculptor) [inline source/Mlos.Python/mlos/OptimizerEvaluationTools/SyntheticFunctions/NestedPolynomialObjective.py:16]: with the exception of the random seed.

Comment (byte-sculptor) [reply source/Mlos.Python/mlos/OptimizerEvaluationTools/SyntheticFunctions/NestedPolynomialObjective.py:16]: so almost identical

File: source/Mlos.Python/mlos/OptimizerEvaluationTools/SyntheticFunctions/NestedPolynomialObjective.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    The value of num_nested_polynomials controls how many polynomials are created. All polynomials are created according to (identical)
+    configs specified by the polynomial_objective_config value.
+
+    The idea here is to provide a more general version of ThreeLevelQuadratic. In ThreeLevelQuadratic we have three two-dimensional,
+    degree two polynomials, and we select between them using the "vertex" parameter.
```
Comment (byte-sculptor) [inline source/Mlos.Python/mlos/OptimizerEvaluationTools/SyntheticFunctions/NestedPolynomialObjective.py:21]: vertex_height

**Review by grlap — APPROVED** [review]

**Pull request merged**